docs/HowToSupportTizenRtProtectedBuild.md

```diff
@@ -0,0 +1,343 @@
+
+	ksram (rwx)      : ORIGIN = 0x20000000, LENGTH = 320K
+	usram (rwx)      : ORIGIN = 0x2004E200, LENGTH = 320K
+```
+
+- In addition to the memory region definition, the developer also needs to define few linker variables in memory.ld file, which will give the start address and size of each region as shown in the sample below:
+
+```
+	__kflash_segment_start__  = ORIGIN(kflash);
+	__kflash_segment_size__   = LENGTH(kflash);
+	__uflash_segment_start__  = ORIGIN(uflash);
+	__uflash_segment_size__   = LENGTH(uflash);
+	__ksram_segment_start__   = ORIGIN(ksram);
+	__ksram_segment_size__    = LENGTH(ksram);
+	__usram_segment_start__   = ORIGIN(usram);
+	__usram_segment_size__    = LENGTH(usram);
+```
+
+#### kernel-space.ld
+- This linker script defines the various 'SECTIONS' in the kernel binary and indicates their placement in one of the memory regions defined in memory.ld.
+- This file can be created by making a copy of the ld.script file which is used for flat build.
+- After making a copy, remove any memory region defenitions and retain only the 'section' definitions in the kernel-space.ld file.
+- Then, the sections assigned to code memory should be replaced with the partitioned 'kernel' code memory and data memory should be replaced with the partitioned 'kernel' data memory.
+- The below snippet shows a sample where text section is assigned to kflash and data section is assigned to ksram region:
+
+```
+SECTIONS
+{
+	.text : {
+		_stext = ABSOLUTE(.);
+		....
+		....
+		_etext = ABSOLUTE(.);
+	} > kflash
+	/* Here in comparison to ld.script for flat build, 'flash' memory region should be replaced with 'kflash' memory region. */
+	....
+	.data : {
+		_sdata = ABSOLUTE(.);
+		....
+		_edata = ABSOLUTE(.);
+	} > ksram
+	/* Here in comparison to ld.script for flat build, 'sram' memory region should be replaced with 'ksram' memory region. */
+}
+```
+
+#### user-space.ld
+- This linker script defines the various 'SECTIONS' in the user binary and indicates their placement in one of the memory regions defined in memory.ld.
+- This script can be created by making a copy of the already created kernel-space.ld script.
+- In this script, declarations of some external functions should be added at the beginning of the script as shown in below snippet. This will be required during linking.
+
+```
+EXTERN(umm_initialize)
+EXTERN(umm_addregion)
+EXTERN(umm_trysemaphore)
+EXTERN(umm_givesemaphore)
+
+EXTERN(malloc)
+EXTERN(realloc)
+EXTERN(zalloc)
+EXTERN(free)
+```
+- Now, the sections assigned to 'kernel' code memory should be replaed with the 'user' code memory and 'kernel' data memory should be replaced with the 'user' data memory.
+- After the build process, when the userspace binary gets created, two additional things are needed to appear as compared to the kernel space binary.
+	- 1. UserSpace Object - This should be placed at the starting of the userspace binary. Add the lines shown in the below snippet before the text section:
```

Comment:
How about removing tab on line 180 and adding enter on line 179?

Reply:
Line 180 is actually a subtopic of line 179. Hence, such an indentation was used.

Reply:
Ok, but it looks so strange in markdown viewer.
so, how about adding tabs on lines 182, 186 and 190-197?

Reply:
Yes, made the required changes. Kindly review it.